Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -152,24 +169,31 @@ private void GenerateCorrelationIdAndAddToDictionary(string ikey, string appId)
         {
             try
             {
+#if !NETCORE
                 SdkInternalOperationsMonitor.Enter();
-
+#endif
                 Uri appIdEndpoint = this.GetAppIdEndPointUri(instrumentationKey);
-
+#if !NETCORE
```

Comment:
In DependencyCollector.NetCore's project.json file I define a buildOption for NETCORE, so it allows me to use NETCORE as a preprocessor definition.